rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -57,7 +59,18 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 Timber.e(e);
             }
         } else if (requestCode == BARCODE_REQUEST_CODE && resultCode == RESULT_CANCELED) {
-            ((RDTJsonFormFragment) widgetArgs.getFormFragment()).setMoveBackOneStep(true);
+            if (data != null && data.getBooleanExtra(Constants.Result.ONESCAN_IS_NOT_INSTALLED, false)) {
```

Comment:
Could we add a test for this functionality?

Reply:
Also, is the boolean in the intent or in the extras of the intent?

Reply:
Its from `intent.putExtras(bundle)`. Can intent have data other than extra?

Reply:
Not sure about the test, I'll try later.